### ISSUE TITLE
Conflicting translations should trigger warning to user

### ISSUE THREAD

**Issue by badrihippo** [sawaliram/sawaliram#219]
When a user tries to translate an item to the same language as an existing translation, they should receive a warning. Some points to consider:

* Should a user receive a warning if the conflicting translation is a draft, or only if it is published?
* If the former, what is the cutoff for the "last updated" period, beyond which the older translation will be considered stalled and not trigger a warning to the user?
* Should the user know who is working on the other translation, so they can coordinate?
* Alternatively, should they be presented with an anonymous form to contact the other translator, so they can coordinate without the translator's personal information being leaked?
* What if users are notified of a existing translation, but also notified about whether that existing translation is published, submitted, or a draft?

Once these points are cleared up, we can work out the logic of how to display the notification. A simple popup will probably be enough.

**Comment by badrihippo** [comment]
@nitinpaul @dchandrika-tifr any ideas?

**Comment by jr-sawaliram** [comment]
It seems useful, even necessary, to inform aspiring translators of an existing translation (published, submitted or draft). Or else could we just remove that item from the translators' view?

Separately, should we warn all draft-writers say every 5 days that they have drafts pending?

**Comment by badrihippo** [comment]
Removing submitted and published items from the view is not so much a problem. We don't need multiple translations; at least not at the moment.

The main "fuzzy area" is whether to display drafts on the list or not. Someone may have created a draft by mistake and forgotten about it, or moved on to something else, in which case there'd have to be some way to still open up those pages for translations. I'd suggest keeping them translatable but showing a warning to the user. How about we say something like "someone was working on this X days ago" so they can decide for themselves if the translation is active or "abandoned"? Alternatively, we could just show the warning for all conflicts where the conflicting translation was last edited (say) 5 or less days ago. @dchandrika-tifr any suggestions?

In any case, the 5 days notification is a good idea. Or maybe it should be 7, to account for people who only work on weekends or on a certain day of the week. But it'll require more work to implement; perhaps we should create a separate issue to discuss that?

**Comment by badrihippo** [comment]
@dchandrika-tifr @jr-sawaliram still waiting on you for suggestions!

**Comment by dchandrika-tifr** [comment]
(apols at the outset, I do not get any notification of these mentions. Looks like they are getting spammed as the github account is connected to my personal gmail id.)

Thanks Badri. These are very valid points to consider (in fact something that came up very early on even with "Answer Questions" management). 

I vote for once in 7-day reminder to the author/translator. I am not sure if translators would be happy collaborating on one item. I see translation is a very personal experience and think that no two translations are the same! To deal with the "fuzzy area" I think it should work if the existing translated items (drafts/submitted/publish) are listed with the user and date-time stamp, and the system not allow an existing translation pair. 

With the mailing system in place (Phase II of the dev project), we can have the users communicate over emails without getting access to mail-id's (right?)

Letting a user decide could be very subjective depending on their "scale of promptness" and since this is a volunteer platform, larger hold-wait times would not be uncommon.

**Comment by nitinpaul** [comment]
My suggestion is to not show items that have a 'submitted/review pending' translation in the listing page, and not worry about 'drafts' at all. With the amount of content there is to be translated and the number of languages to translate to, it doesn't seem very likely that some other translator will submit a translation before you, while you still have a draft for it. We don't know how much time it will take on average to translate an answer, or how many translators will be working in tandem. Let's solve this once this is a problem!

**Comment by badrihippo** [comment]
One more complication, that I realised while discussing with @dchandrika-tifr. We can't unilaterally hide a translated item, because while that article may be translated into one language (eg. Hindi) it may still be open for translation in another (eg. Bengali). If we hide it when the Hindi translation comes out, it'll never get translated to Bengali!

Two options:
1. List most items for translation, hiding only the ones that have been translated into *all the selected languages*. By "selected" I mean chosen in the search filter menu on the side. And then, display the "This item already has a translation in the selected language" message in the language selection page itself. In this case, it would make sense to preserve the intermediate language selection page, as a place to display such messages.
2. Ask the person to choose the target language in the search itself. That means the "Translate From [source language] -> To [target language]" selector shows on the search page itself, and items are only shown if they are both (a) written in the source language and (b) do not have a translation for the target language. So only items that we want to be translated from [source language] to [target language]. This way, we can skip the intermediate prompt since the source and target language are already chosen at the outset and there's no need for intermediate notifications.

In both these cases, we have to decide whether to include "already has a draft" and "already under review" in the exclusion criteria. I think we can include them for now, as Nitin suggested, and then open up to the more flexible "intelligent warning" system when enough translations make it necessary.

**Comment by badrihippo** [comment]
For context, it was @jr-sawaliram's suggestion that we use the intermediate "Translate Content" page, shown below, as the place to display "translation already exists" notifications.

![image](https://user-images.githubusercontent.com/7288084/78558793-da08e200-7830-11ea-9fd3-ca20b2e345b3.png)

However, @nitinpaul wanted the screen eliminated entirely (it wasn't part of the original design, and it adds an extra step for the user to go through to begin translating, which results in more friction for the user).

**Comment by nitinpaul** [comment]
> We can't unilaterally hide a translated item, because while that article may be translated into one language 

That's not what I meant by hiding items that have a translation. Anyway, option 2 here seems intuitive.
